src/main/java/com/expidevapps/android/measurements/support/v4/fragment/BaseEditTrainingDialogFragment.java

```diff
@@ -192,4 +236,29 @@ protected void updateTrainingDateLabel(@Nullable final LocalDate date) {
                     .setText(date != null ? DateFormat.getDateInstance(DateFormat.SHORT).format(date.toDate()) : "");
         }
     }
+
+    @Optional
+    @OnClick(R.id.show_training)
+    void onShowTrainingTap() {
+        mTrainingData.setVisibility(View.VISIBLE);
+        mStagesData.setVisibility(View.GONE);
+    }
+
+    @Optional
+    @OnClick(R.id.show_stages)
+    void onShowTrainingStagesTap() {
+        mStagesData.setVisibility(View.VISIBLE);
+        mTrainingData.setVisibility(View.GONE);
+    }
+
+    @Optional
+    @OnClick(R.id.et_new_completion_date)
+    void onNewTrainingCompletionDateClick() {
```

Comment:
This is currently pulling from the training date, not the date for the new training completion.